avm/res/storage/storage-account/main.bicep

```diff
@@ -69,10 +69,7 @@ param privateEndpoints privateEndpointType
 param managementPolicyRules array?
 
 @description('Required. Networks ACLs, this value contains IPs to whitelist and/or Subnet information. If in use, bypass needs to be supplied. For security reasons, it is recommended to set the DefaultAction Deny.')
-param networkAcls networkAclsType = {
-  bypass: 'AzureServices'
-  defaultAction: 'Deny'
-}
+param networkAcls networkAclsType
```

Comment:
Intended removal?

Reply:
I know there is a related issue open - yet it may have sneaked it's way into this PR.

Reply:
> Intended removal?

Initially, it was not intended, however this default value causes a Bicep error and breaks the Set-AVMModule execution. And because the parameter description describes it as "Required", I assumed that there should no default here.

![image](https://github.com/Azure/bicep-registry-modules/assets/20225789/efcdc0e6-5381-4d35-9b7e-b7515e7bde61)

Reply:
Fair, but it does change the intend. So what we should figure out (with @fblix) is how to handle it properly. Our options:
- Move the default away from the parameter to where it is used (like we did for diagnostics etc.)
  ```bicep
  networkAcls: !empty(networkAcls) ? {
      resourceAccessRules: networkAcls.?resourceAccessRules
      bypass: networkAcls.?bypass
      defaultAction: networkAcls.?defaultAction
      virtualNetworkRules: networkAcls.?virtualNetworkRules
      ipRules: networkAcls.?ipRules
  } : {
      bypass: 'AzureServices'
      defaultAction: 'Deny'
  }
  ```
- Make the UDT not nullable. If the parameter anyways has a default, making it's typo to nullable is an easy change.

Reply:
@fblix do you have any preference?

Reply:
@fblix I excluded the Storage Account from this module. Please update the PE implementation separately.